substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/Containers.java

```diff
@@ -220,5 +220,10 @@ public void duringSetup(DuringSetupAccess access) {
         RuntimeClassInitializationSupport classInitSupport = ImageSingletons.lookup(RuntimeClassInitializationSupport.class);
         classInitSupport.initializeAtRunTime("com.oracle.svm.core.containers.cgroupv1.CgroupV1Subsystem", "for cgroup support");
         classInitSupport.initializeAtRunTime("com.oracle.svm.core.containers.cgroupv2.CgroupV2Subsystem", "for cgroup support");
+        // JDK platform mirror classes need to get re-initialized at runtime so as to update to the
+        // correct path of the cgroup interface files stored in controllers. This code-path is being used for example via
+        // JFR tracing.
+        classInitSupport.rerunInitialization("jdk.internal.platform.cgroupv1.CgroupV1Subsystem", "for cgroup support");
```

Comment:
It looks like `jdk.internal.platform.CgroupV1MetricsImpl` will need some special handling as well. I wonder if we indeed need to re-run initialization on these classes or if it would suffice to reset the `INSTANCE` field value.

The error from the CI with the current approach is:
```
 com.oracle.svm.core.util.UserError$UserException: No instances of jdk.internal.platform.cgroupv1.CgroupV1Subsystem are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=jdk.internal.platform.cgroupv1.CgroupV1Subsystem.
Detailed message:
Trace: Object was reached by
  reading field jdk.internal.platform.CgroupMetrics.subsystem of constant 
    jdk.internal.platform.CgroupV1MetricsImpl@38b7a16c: jdk.internal.platform.CgroupV1MetricsImpl@38b7a16c
  reading static field jdk.jfr.internal.instrument.JDKEvents.containerMetrics
    at jdk.jfr.internal.instrument.JDKEvents.emitContainerIOUsage(JDKEvents.java:242)
  parsing method jdk.jfr.internal.instrument.JDKEvents.emitContainerIOUsage(JDKEvents.java:242) reachable via the parsing context
    at jdk.jfr.internal.instrument.JDKEvents$$Lambda$69/0x00000007c00e7aa8.run(Unknown Source)
    at java.lang.Shutdown.runHooks(Shutdown.java:130)
    at java.lang.Shutdown.exit(Shutdown.java:173)
    at com.oracle.svm.core.graal.snippets.CEntryPointSnippets.initializeIsolate(CEntryPointSnippets.java:314)
```

Reply:
Thanks, good point. That might work.

Reply:
Updated the patch with resetting various statics instead. Lets see what CI says to this.